### ISSUE TITLE
CPU is consumed when paused and minimized in the dock

### ISSUE THREAD

**Issue by low-batt** [iina/iina#3537]
**System and IINA version:**

- macOS 12.0.1
- IINA 1.2.0 Build 129

**Expected behavior:**

IINA follows the best practices in the [Energy Efficiency Guide for Mac Apps](https://developer.apple.com/library/archive/documentation/Performance/Conceptual/power_efficiency_guidelines_osx/BestPractices.html#//apple_ref/doc/uid/TP40013929-CH38-SW1). In the case at hand:
> Ensure your app is absolutely idle when not in use

**Actual behavior:**

IINA continues to consume a small amount of CPU when a video window is paused and minimized to the dock. This capture was done on a M1 MacBook Pro with one local video file paused and the window minimized to the dock::

```text
low-batt@gag ~$ while true; do ps -c -o 'comm,%cpu,cputime' -u low-batt | grep IINA | grep -v grep; sleep 5; done
IINA               0.5   0:01.10
IINA               0.6   0:01.14
IINA               0.6   0:01.18
```

CPU use increases with the number of IINA windows opened and paused. In this case there were 3 windows paused and minimized in the dock:

```text
low-batt@gag ~$ while true; do ps -c -o 'comm,%cpu,cputime' -u low-batt | grep IINA | grep -v grep; sleep 5; done
IINA               1.2   0:04.13
IINA               1.1   0:04.22
IINA               1.1   0:04.30
```

**Steps to reproduce:**

Open several local video files in IINA, pause them, minimize the windows to the dock, use Activity Monitor or ps to confirm IINA is still consuming CPU time.

- [ ] MPV does not have this problem.

MPV also has this problem, but MPV is not causing the IINA issue.
I already have tracked down what is consuming CPU.
I will shortly post a pull request with proposed changes to address this issue.

**How often does this happen?**

Every time.

**Comment by low-batt** [comment]
### Analysis

`PlayerCore.createSyncUITimer` creates a timer that calls `syncUITime` to keep the UI synchronized with mpv. This timer is still running when playback is paused.

`VideoView.startDisplayLink` creates a [CVDisplayLink](https://developer.apple.com/documentation/corevideo/cvdisplaylink-k0k) thread. This thread is still running when playback is paused.

### Fixing

The commit in the pull request:
- Changes `PlayerCore.pause` to invalidate the timer and stop the display link thread
- Changes `PlayerCore.resume` to create the synchronization timer and start the display link

With 3 windows paused and minimized in the dock after these changes have been applied:
```text
low-batt@gag ~$ while true; do ps -c -o 'comm,%cpu,cputime' -u low-batt | grep IINA | grep -v grep; sleep 5; done
IINA               0.0   0:03.04
IINA               0.0   0:03.04
IINA               0.0   0:03.04
```

**Comment by pleasejustgo** [comment]
What is going on with the devs? Are they ever going to continue? Why aren't they allowing someone else who wants to contribute who is free to make changes and release an update? wtf?

**Comment by low-batt** [comment]
Closing. Fixed in IINA 1.3.0.
